.github/workflows/pr.yml

```diff
@@ -12,11 +17,15 @@ jobs:
   build-app:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - name: set up JDK 17
-        uses: actions/setup-java@v1
+      - uses: actions/checkout@v4
+      - name: set up JDK 21
+        uses: actions/setup-java@v4
```

Comment:
I always wondered if we needed that given that the images [come preinstalled with Java 21](https://github.com/actions/runner-images/blob/ubuntu24/20250209.1/images/ubuntu/Ubuntu2404-Readme.md?plain=1#L113)

Reply:
The action sets the JDK version that you want to use (multiple versions are preinstalled). From what I can see in the runner logs, it looks like it just uses the version that is preinstalled in the image when available.

Reply:
Nice!